### PR TITLE
[UAP] Roles and Collections cont'd

### PR DESCRIPTION
**Pull request by MCGallaspy**
## Summary

Building on #22, continues to implement Roles and Collections functionality. Next up is to implement permissions checking. Will rebase when #22 is merged.

## TODO

- [x] Have tests been written for the new code?
- [x] Has documentation been written/updated?

## Reviewer guidance

Mainly implements model methods, see `CollectionRelatedObjectTestCase` and `CollectionRemovalTestCase`.

## Documentation

The [tech specs](https://docs.google.com/document/d/1BcOSxhoAOZGKjg9e7C6PpeQfHj9zkDqrKIk1WrbLsDU/edit#heading=h.fzctnhrv2u2t).

### CODE REVIEW

File: kolibri/auth/test/test_collections_and_roles.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        self.f.delete()
+        self.assertEqual(Collection.objects.count(), 0)
+        self.assertEqual(Role.objects.count(), 0)
+
+
+class CollectionRelatedObjectTestCase(TestCase):
+    def setUp(self):
+        users = self.users = [FacilityUser.objects.create(username="foo%s" % i) for i in range(0, 10)]
+        self.lg = LearnerGroup.objects.create()
+        self.lg.add_learners(users[0:5])
```
Comment (benjaoming) [inline kolibri/auth/test/test_collections_and_roles.py:86]: nice clean interface! :+1:

File: kolibri/auth/migrations/0002_auto_20160203_1516.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
```
Comment (benjaoming) [inline kolibri/auth/migrations/0002_auto_20160203_1516.py:1]: Let's btw add the `fix-encoding-pragma` pre-commit hook?

Comment (benjaoming) [inline kolibri/auth/migrations/0002_auto_20160203_1516.py:1]: https://github.com/pre-commit/pre-commit-hooks#hooks-available

File: kolibri/auth/test/test_collections_and_roles.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
+from django.test import TestCase
```
Comment (benjaoming) [inline kolibri/auth/test/test_collections_and_roles.py:3]: Missing py3 `__future__`

File: kolibri/auth/test/test_users.py [hunk 4]
```diff
@@ -2,8 +2,7 @@
 
 from django.test import TestCase
 
-# Importing user models here results in a circular import... you should use get_user_model, but then there's no
-# way to get the proxy models as well. So just import at runtime.
+from kolibri.auth.models import FacilityUser, DeviceOwner, BaseUser, KolibriValidationError
```
Comment (benjaoming) [inline kolibri/auth/test/test_users.py:5]: Nice, this is the stuff that wasn't working when the django env wasn't setup before running the tests?

File: pytest.ini [hunk 5]
```diff
@@ -1,2 +1,3 @@
 [pytest]
 testpaths = test/ kolibri/
+DJANGO_SETTINGS_MODULE = kolibri.deployment.default.settings.base
```
Comment (benjaoming) [inline pytest.ini]: Let's call this `kolibri.deployment.default.settings.test` ?

**Comment by benjaoming** [comment]
Minor details commented, please do merge and rebase #28 at will.

File: kolibri/auth/test/test_users.py (hunk 4 above)
Comment (MCGallaspy) [inline kolibri/auth/test/test_users.py:5]: Right, but I didn't realize that was the error initially.

File: kolibri/auth/migrations/0002_auto_20160203_1516.py (hunk 2 above)
Comment (MCGallaspy) [inline kolibri/auth/migrations/0002_auto_20160203_1516.py:1]: I can't seem to get this pre-commit hook to work

Comment (benjaoming) [inline kolibri/auth/migrations/0002_auto_20160203_1516.py:1]: Yeah, it was weird... I had to disable it, then autoreload, then enable it again. Not sure what's up. But it works here.

**Comment by MCGallaspy** [comment]
Just to be clear, I'd like someone *other than me* to hit the merge button. This is something I intended as part of our [governance model](https://docs.google.com/document/d/1Hebvda2YIMed__MDDVrg1iJav2YHK4zYEXJ59ITmCcE/edit), even though it wasn't explicit. I'm now adding a suggestion to make this explicit.

**Comment by benjaoming** [comment]
Sorry I didn't put an explicit comment, but this is the PR that I made a PR for in your repo, @MCGallaspy ..which is what made me not merge it immediately.

**Comment by MCGallaspy** [comment]
@benjaoming your PR to my PR raises a test error for me. I'm merging this for now and we can address that separately.

**Pull request merged**